pA vertex smearing update to match measured beamspots

mandrenguyen opened the pull request:
Vtx smearing parameters are updated to match 5 TeV and 8 TeV beamspots measured by @jrcastle 
The pixel barycenter shift between data and MC is taken into account. 
Variants of the vtx smearing that also boost the center-of-mass to do asymmetric collisions in MC are updated as well. 
This PR will be backported to 80X.  Should I also put it in 81X?

mandrenguyen committed:
Added 5 TeV pA beamspot and placeholder for 8 TeV

mandrenguyen committed:
Updated pA 8 TeV relval wf, number 281

mandrenguyen committed:
Updated 8 TeV vtx smearing parameters

cmsbuild commented:
A new Pull Request was created by  @mandrenguyen  for CMSSW_9_0_X.

It involves the following packages:

Configuration/PyReleaseValidation
Configuration/StandardSequences
GeneratorInterface/HiGenCommon
IOMC/EventVertexGenerators


@civanch, @perrozzi, @franzoni, @mdhildreth, @fabozzi, @cmsbuild, @srimanob, @thuer, @govoni, @hengne, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @mkirsano, @makortel, @felicepantaleo, @cbaus, @GiacomoSguazzoni, @rovere, @VinInn, @Martin-Grunewald, @agrohsje, @dgulhan this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: a0a6dc903f5c45907c61028b8149c31fa500ff7a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16786/16666/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16786/16666/summary.html

The workflows 1003.0, 1001.0, 1000.0, 140.53, 136.731, 4.22 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

mandrenguyen committed:
Mistakenly took 5 TeV numbers before pixel barycenter shift -- fixed

cmsbuild commented:
Pull request #16786 was updated. @civanch, @perrozzi, @franzoni, @mdhildreth, @fabozzi, @cmsbuild, @srimanob, @thuer, @govoni, @hengne, @davidlange6 can you please check and sign again.

mandrenguyen committed:
sigma z updated

cmsbuild commented:
Pull request #16786 was updated. @civanch, @perrozzi, @franzoni, @mdhildreth, @fabozzi, @cmsbuild, @srimanob, @thuer, @govoni, @hengne, @davidlange6 can you please check and sign again.

davidlange6 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 860124a31302ef520962033649ea3d8a94ecb4c2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16786/16701/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16786/16701/summary.html

The workflows 1003.0, 1001.0, 1000.0, 140.53, 136.731, 4.22 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

civanch commented:
+1
@mandrenguyen , is it OK for a backport of this PR to remove a scenario? For me this is fine for 9_0_X. Also likely you need make an extra PR for 8_1_X.

mandrenguyen commented:
@civanch Thanks, I'll make an 81X PR as well then.

davidlange6 commented:
Seems unlikely that we will use 81x for pPb. I think that work can be skipped

Sent from my iPad

On Dec 5, 2016, at 5:25 AM, mandrenguyen <notifications@github.com<mailto:notifications@github.com>> wrote:


@civanch<https://github.com/civanch> Thanks, I'll make an 81X PR as well then.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/16786#issuecomment-264830289>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEzyw-fdLHzKp4GBg_aZTfkTPcyfhAJQks5rE_STgaJpZM4K-pvW>.

mandrenguyen commented on Configuration/PyReleaseValidation/python/relval_steps.py:
```diff
@@ -601,7 +601,7 @@ def genS(fragment,howMuch):
 
 ## pPb Run2
-step1PPbDefaultsUp15={'--beamspot':'Run2PPbBoost','--conditions':'auto:run2_mc_pa','--eventcontent':'RAWSIM', '--era':'Run2_2016_pA'}
+step1PPbDefaultsUp15={'--beamspot':'RealisticPPbBoost8TeV2016Collision','--conditions':'auto:run2_mc_pa','--eventcontent':'RAWSIM', '--era':'Run2_2016_pA'}
```
@davidlange6 :  Yes, @jrcastle is taking charge of this.  We are currently reconstructing a sample with a temporary BS, based on this vertex smearing, to evaluate the reco BS.  Once that's done we'll update the GT, as well as the autocond's.  We prefer to do this in two steps, since the GEN-SIM can already start once the vertex smearing is integrated.

mandrenguyen replied:
@davidlange6 That's too complicated IMO, as the new pPb vertex smearing is now what's linked from the boost tool.  I would have to undo that, which is the entire point of this PR.  If we must really keep a consistent relval, then I'll just update the autoconditions with the GT when it's ready.

mandrenguyen committed:
Update relval_steps.py

cmsbuild commented:
Pull request #16786 was updated. @civanch, @perrozzi, @franzoni, @mdhildreth, @fabozzi, @cmsbuild, @srimanob, @thuer, @govoni, @hengne, @davidlange6 can you please check and sign again.

mandrenguyen commented:
On second thought, I simply unboosted the workflow to use:
Realistic50ns13TeVCollision
Not ideal, but at least self-consistent.
It's getting urgent to get the MC running, so I'd prefer not to wait in 
case there's a delay with the GT.

On 12/8/16 4:28 PM, David Lange wrote:
>
> *@davidlange6* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In Configuration/PyReleaseValidation/python/relval_steps.py 
> <https://github.com/cms-sw/cmssw/pull/16786>:
>
> > @@ -601,7 +601,7 @@ def genS(fragment,howMuch):
>   steps['AMPT_PPb_5020GeV_MinimumBias']=merge([{'-n':10},step1PPbDefaults,genS('AMPT_PPb_5020GeV_MinimumBias_cfi',Kby(9,100))])
>   
>   ## pPb Run2
> -step1PPbDefaultsUp15={'--beamspot':'Run2PPbBoost','--conditions':'auto:run2_mc_pa','--eventcontent':'RAWSIM', '--era':'Run2_2016_pA'}
> +step1PPbDefaultsUp15={'--beamspot':'RealisticPPbBoost8TeV2016Collision','--conditions':'auto:run2_mc_pa','--eventcontent':'RAWSIM', '--era':'Run2_2016_pA'}
>
> Hi @mandrenguyen <https://github.com/mandrenguyen> - then lets 
> separate the relval test from the gen-sim - if you revert this one 
> change then I think we don't break the relval config while this GT is 
> being put together
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/16786>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AEykWjyzphDEnDE6cgLycX4oJUHmoww-ks5rGCI1gaJpZM4K-pvW>.
>


-- 
-----------------------
Matthew Nguyen
Chargé de Recherche
LLR-Ecole Polytechnique
91128 Palaiseau FRANCE
+33 1 69 33 55 65
-----------------------

perrozzi commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 33acc8c8769505519aef57cc8182d372f9f689a6
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16786/16887/summary.html

cmsbuild commented:
Comparison job queued.

civanch commented:
+1

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16786/16887/summary.html

mmusich commented on Configuration/PyReleaseValidation/python/relval_steps.py:
```diff
@@ -601,7 +601,7 @@ def genS(fragment,howMuch):
 
 ## pPb Run2
-step1PPbDefaultsUp15={'--beamspot':'Run2PPbBoost','--conditions':'auto:run2_mc_pa','--eventcontent':'RAWSIM', '--era':'Run2_2016_pA'}
+step1PPbDefaultsUp15={'--beamspot':'Realistic50ns13TeVCollision','--conditions':'auto:run2_mc_pa','--eventcontent':'RAWSIM', '--era':'Run2_2016_pA'}
```
@mandrenguyen @jrcastle in light of what is requested [https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2775.html](here), when changing the RECO BeamSpot to `BeamSpotObjects_Realistic8TeVPACollision2016_v1` in the `auto:run2_mc_pa` Global Tag, shouldn't be the default vertex smearing changed as well?
   * would you mind if I update the autoCond as part of this PR adding commits to yours?
   * can you clarify at that point which is the preferred `--beamspot` option you'd like to have in relvals? is it `RealisticPPbBoost8TeV2016Collision` or `RealisticPbPBoost8TeV2016Collision` ?
Thanks,

M. 

@franzoni @arunhep

mmusich commented:
> It's getting urgent to get the MC running, so I'd prefer not to wait in case there's a delay with the GT.

after 9h (whoa!) here's your GT: 90X_mcRun2_pA_v1
diff with the previous one: https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/90X_mcRun2_pA_v0/90X_mcRun2_pA_v1

based on https://github.com/cms-sw/cmssw/pull/16786#discussion-diff-90903565R604 I see that the preference is given to the p-Pb boost version.
I'll let you revert that and updated autoCond as well.
FYI: you can further minimize the delay by building yourselves GT candidates: (cf. https://indico.cern.ch/event/507993/contributions/2020446/attachments/1252208/1847694/talk4_-_hands_on_tutorial.pdf)

mandrenguyen commented:
@mmusich Thank you for the quick work.  I will update the PR tomorrow (Saturday)

mandrenguyen committed:
Update relval_steps.py

Reverting to boosted beamspot with pA 8 TeV vertex smearing.  A separate PR updating the pA autoconditions to use the GT with the corresponding reco BS tag will be submitted momentarily.

cmsbuild commented:
Pull request #16786 was updated. @civanch, @perrozzi, @franzoni, @mdhildreth, @fabozzi, @cmsbuild, @srimanob, @thuer, @govoni, @hengne, @davidlange6 can you please check and sign again.

civanch commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Comparison job queued.

mmusich commented:
hold
I would say this PR lacks for consistency the commit in #16961

cmsbuild commented:
Pull request has been put on hold by  @mmusich
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

mmusich commented:
unhold
OK - after having clarified with @mandrenguyen I think we can let the two PRs strategy go ahead, provided there are no objection from David @davidlange6 to merge them shortly one after the other.

mmusich commented on IOMC/EventVertexGenerators/python/VtxSmearedParameters_cfi.py:
```diff
@@ -624,6 +624,32 @@
+    TimeOffset = cms.double(0.0),
+    X0 = cms.double(0.0889),
+    Y0 = cms.double(0.1820),
+    Z0 = cms.double(1.6066)    
```
```
-----------------------------------------------------
              Beam Spot Data

 Beam type    = 2
       X0     = 0.0889243 +/- 1.71057e-05 [cm]
       Y0     = 0.181999 +/- 1.70107e-05 [cm]
       Z0     = 1.27369 +/- 0.0942173 [cm]
 Sigma Z0     = 6.81368 +/- 0.0666215 [cm]
 dxdz         = -1.28221e-06 +/- 2.20256e-06 [radians]
 dydz         = 1.96211e-06 +/- 2.18636e-06 [radians]
 Beam Width X = 0.00598025 +/- 6.09373e-05 [cm]
 Beam Width Y = 0.00608967 +/- 6.09373e-05 [cm]
 Emittance X  = 0 [cm]
 Emittance Y  = 0 [cm]
 Beta star    = 0 [cm]
-----------------------------------------------------
```
also for [BeamSpotObjects_Realistic5TeVPACollision2016_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/tags/BeamSpotObjects_Realistic5TeVPACollision2016_v1), the fit value of  SigmaZ0 seems to be a bit overestimated.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16786/16934/summary.html

jrcastle commented on IOMC/EventVertexGenerators/python/VtxSmearedParameters_cfi.py:
```diff
@@ -624,6 +624,32 @@
+    X0 = cms.double(0.0836),
+    Y0 = cms.double(0.1837),
+    Z0 = cms.double(1.3577)    
+)
```
@mmusich, this overestimation of SigmaZ0 is a common effect seen when fitting the reco BS fo pPb.  This was observed a few months back when performing studies on whether a separate Pbp tag would be needed.  The SigmaZ0 is consistently larger in the reco BS fit.

As far as alignment is concerned, I pulled a stable PCL alignment from 5 and 8 TeV respectively when proposing the vertex smearing values.

mmusich replied:
@jrcastle, I was more concerned about the alignment used for the final fit in the MC samples. In this [request](https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/2775.html) you exaplain clearly the points:

1. An average beamspot was computed with Prompt RECO data using all
available runs for a given collision energy. The DCS JSON was also used
to filter lumis.

2. The centroid of this average beamspot was then translated to account
for the barycenter of the pixel tracker. See [1].

3. A new vertex smearing parameter set was included in CMSSW and a
"fake" payload was uploaded to the conditions database, each containing
the new, shifted beamspot. See [2].

4. The new vertex smearing parameters and fake payload were used to
simulate and reconstruct 100k MinBias pPb events.

5. A new fit of the beamspot was made to the MC data and uploaded to the
conditions database. This payload is the RECO BS we would like included
in a GT.

I would assume that the fake payload used in point 4. was closer to the vertex smearing than the final one in point 5.  Which conditions were used to reconstruct these 100k MinBias pPb events? I am asking because artificial of the BS sigmaZ is a known effect of exansion/shrinkage of the tracker alignment z-scale.

jrcastle replied:
@mmusich, my apologies on the mixup.  We used the default alignment in the GT 80X_mcRun2_pA_v1.  Looking that up on conddb shows that the alignment record for that particular tag is TrackerAlignment_Asymptotic_Run2016_v2_mc.

civanch commented:
+1

davidlange6 commented:
Hi @mandrenguyen @mmusich - could we get one PR with everything?

mmusich commented:
Hi @mandrenguyen I have the branch ready for addressing https://github.com/cms-sw/cmssw/pull/16786#issuecomment-266377549:

https://github.com/cms-sw/cmssw/compare/CMSSW_9_0_X...cms-AlCaDB:pAVtxSmearingAndAutoCond_90X?expand=1

do you want to use this one (and close this PR) ?

mandrenguyen commented:
Superseded by #16974
